features/bootstrap/DatasetViewContext.php

```diff
@@ -417,14 +417,13 @@ public function iHaveSetPagesizeToOn($arg1, $arg2)
     }
 
     /**
-     * @Then There should be a meta-tag element has attribute :arg1 with value :arg2 and attribute :arg3 with value :arg4
+     * @Then There should be a meta-tag has attribute :arg1 with value :arg2 and attribute :arg3 with value :arg4
      */
-    public function thereShouldBeMetaElementWithDoi($arg1, $arg2, $arg3, $arg4)
+    public function thereShouldBeMetaTagWithNameAndWithContent($arg1, $arg2, $arg3, $arg4)
     {
-        $metaNode = $this->minkContext->getSession()->getPage()->find('css', "meta[$arg1='$arg2']");
-        if( $metaNode->hasAttribute($arg1) && $metaNode->hasAttribute($arg3)) {
-            PHPUnit_Framework_Assert::assertEquals($arg2, $metaNode->getAttribute($arg1));
-            PHPUnit_Framework_Assert::assertEquals($arg4, $metaNode->getAttribute($arg3));
-        }
+        $metaNode = $this->minkContext->getSession()->getPage()->find('xpath', "//meta[@$arg1='$arg2' and @$arg3='$arg4']");
+        PHPUnit_Framework_Assert::assertNotNull($metaNode);
+        PHPUnit_Framework_Assert::assertEquals($arg2, $metaNode->getAttribute($arg1));
+        PHPUnit_Framework_Assert::assertEquals($arg4, $metaNode->getAttribute($arg3));
```

Comment:
Hi @rija 
I have updated the function using `xpath` which could hold two attributes at the same time. In fact, I tried the same way and put it into `('css', "meta[$arg1='$arg2' $arg3='$arg4']")`, but it could not work😢 

In this function, I have added 2 more parameters, `$arg1` and `$arg3`,  which may make it more reusable for future use. And I have add 2 more assertions to further confirm that the returned value equals to what we expected.

Is it redundant to add the extra parameters and assertions?

Reply:
Hi @kencho51,

When you work on a functionality you should always focus on making it working according to specifications first with the simplest code that is readable by the future you and others.
Then you can assess the state of the code and see if there's anything worth improving.

Remember the scenario steps have to be understood by anyone in the business. Acceptance tests is in effect a business contract between the product owner and the development team (hence the name),
so when writing the steps, never sacrifice conciseness and readability for re-usability. 

If you are looking for potential for re-use, always start thinking of a private function that can be used by multiple steps (you've already done that in a previous task), if opportunity is not there, leave it be.

So as it stands now,

```
And There should be a meta-tag has attribute "name" with value "citation_doi" and attribute "content" with value "10.5072/100002"
```

is not a good sentence for a business contract between the product owners and the development team.

Beside, look at the definition of a meta-tag in the HMTL specs, there is not a large number of attributes that ``<meta>`` elements can takes, so the effort of trying to make the step generic with  the attributes key as variables is not worth losing the readability and precision.

We can always change our mind later if and only if the situation calls for it.

To answer your question about the assertions, yes they are redundant because the fact that the xpath return a non-null object is enough proof since we specified an xpath for a <meta> with the given name and content. If the meta-tag is not there or didn't have the specified name AND content, the query would return a null object. Therefore it's enough to assert the query return something that's not null.